packages/components/timepicker/src/components/osds-timepicker/core/controller.ts

```diff
@@ -0,0 +1,42 @@
+import type { OsdsTimepicker } from '../osds-timepicker';
+import { ODS_TIMEZONE, ODS_TIMEZONES } from '../constants/timezones';
+import { ODS_TIMEZONES_PRESET } from '../constants/timezones-preset';
+
+class OdsTimepickerController {
+
+  constructor(private readonly component: OsdsTimepicker) {
+  }
+
+  checkSeconds(withSeconds: boolean) {
+    if (withSeconds === false && (this.component.input as HTMLInputElement)?.value.match(/:/g)?.length === 2) {
+      const inputValue = (this.component.input as HTMLInputElement).value.split(':');
+      (this.component.input as HTMLInputElement).value = `${inputValue[0]}:${inputValue[1]}`
+    } else if (withSeconds === true && (this.component.input as HTMLInputElement)?.value.match(/:/g)?.length === 1) {
+      (this.component.input as HTMLInputElement).value = `${(this.component.input as HTMLInputElement).value}:00`
```

Comment:
TBD: I think we should stop this back and forth between component and controller:
```
(component)checkSeconds =>
  => call (controller) checkSeconds
    => use (component)input
```
Plus in this case we set the value of the component element, in the controller, in a check function.


Would be more clear (and easier to unit test) to have something like:
```
(component)
  @Watch('withSeconds')
  @Watch('value')
  onSecondsChange(withSeconds: boolean) {
    const value = this.controller.formatValue(this.input.value, withSeconds);

    if (value) {
      this.input.value
    }
  }

(controller)
formatValue(value: string, withSeconds: boolean): string {
  if ... {
     ...
     return ...
  } else if ... {
     ...
     return ...
  }
  return '';
}
```